scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -382,12 +382,22 @@
             :to-make-room-for pending-job)]
     [state nil]))
 
+(defn reserve-hosts! [state preemption-decisions]
+  (let [multiple-tasks (filter #(< 1 (count (:task %))) preemption-decisions)
+        reservations (into {} (map (fn [d] [(:job/uuid (:to-make-room-for d))
+                                            (:hostname d)])
+                                   multiple-tasks))]
+    (swap! state (fn [{:keys [jobs-launched]}]
+                   {:reservations (apply dissoc reservations jobs-launched)
+                    :jobs-launched #{}}))))
```

Comment:
Why is the function clearing out `jobs-launched`? Maybe we can clarify with a comment?

Reply:
Nitpick: lexicographic ordering.

Reply:
Added a comment. These are in my opinion ordered correctly.